crates/zksync/core/src/convert.rs

```diff
@@ -150,17 +150,23 @@ impl ConvertSignature for ZkSignature {
     }
 
     fn to_alloy(self) -> AlloySignature {
-        AlloySignature::from_rs_and_parity(self.r.to_ru256(), self.s.to_ru256(), self.v).unwrap()
+        let parity = alloy_primitives::Parity::try_from(self.v).unwrap();
+        AlloySignature::from_scalars_and_parity(
+            self.r.to_b256(),
+            self.s.to_b256(),
+            parity.y_parity(),
+        )
     }
 }
 
 impl ConvertSignature for AlloySignature {
     fn to_ethers(self) -> ZkSignature {
-        let v = self.v();
+        let parity = alloy_primitives::Parity::from(self.v());
```

Comment:
Shouldn't this already be of the type `Parity`?

Reply:
unfortunately in the `PrimitiveSignature` it's just a bool